packages/console/src/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import classNames from 'classnames';
+import { nanoid } from 'nanoid';
+import React, { InputHTMLAttributes, ReactNode, useState } from 'react';
+
+import * as styles from './index.module.scss';
+
+const Check = () => (
+  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <rect x="1" y="1" width="18" height="18" rx="4" fill="#5D34F2" />
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M8.31476 13.858L5.13295 10.441C4.95568 10.253 4.95568 9.947 5.13295 9.757L5.77568 9.074C5.95295 8.886 6.24113 8.886 6.4184 9.074L8.63657 11.466L13.5811 6.141C13.7584 5.953 14.0465 5.953 14.2238 6.141L14.8665 6.825C15.0438 7.013 15.0438 7.32 14.8665 7.507L8.95748 13.858C8.78021 14.046 8.49203 14.046 8.31476 13.858Z"
+      fill="white"
+    />
+  </svg>
+);
+
+const CheckDisabled = () => (
+  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <rect x="1" y="1" width="18" height="18" rx="4" fill="#C4C7C7" />
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M8.31476 13.858L5.13295 10.441C4.95568 10.253 4.95568 9.947 5.13295 9.757L5.77568 9.074C5.95295 8.886 6.24113 8.886 6.4184 9.074L8.63657 11.466L13.5811 6.141C13.7584 5.953 14.0465 5.953 14.2238 6.141L14.8665 6.825C15.0438 7.013 15.0438 7.32 14.8665 7.507L8.95748 13.858C8.78021 14.046 8.49203 14.046 8.31476 13.858Z"
+      fill="white"
+    />
+  </svg>
+);
+
+const Uncheck = () => (
+  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <rect x="2" y="2" width="16" height="16" rx="3" stroke="#8E9192" strokeWidth="2" />
+  </svg>
+);
+
+const UncheckDisabled = () => (
+  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <rect
+      x="2"
+      y="2"
+      width="16"
+      height="16"
+      rx="3"
+      fill="#EFF1F1"
+      stroke="#C4C7C7"
+      strokeWidth="2"
+    />
+  </svg>
+);
+
+type Props = Omit<InputHTMLAttributes<HTMLInputElement>, 'type'> & {
+  label?: ReactNode;
+};
+
+const Checkbox = ({ label, disabled, ...rest }: Props) => {
+  const [id] = useState(nanoid());
+
+  return (
+    <div className={styles.checkbox}>
+      <input id={id} type="checkbox" disabled={disabled} {...rest} />
+      <span className={classNames(styles.icon, styles.check)}>
+        <Check />
+      </span>
+      <span className={classNames(styles.icon, styles.uncheck)}>
+        <Uncheck />
+      </span>
+      <span className={classNames(styles.icon, styles.uncheckDisabled)}>
+        <UncheckDisabled />
+      </span>
+      <span className={classNames(styles.icon, styles.checkDisabled)}>
+        <CheckDisabled />
+      </span>
```

Comment:
Not sure if this helps but feels like it is an easier way to implement this. Following is how I did in UI:

Icon.svg
```svg
<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
  <symbol width="18" height="18" viewBox="0 0 18 18" id="unchecked">
      <circle cx="9" cy="9" r="8" stroke="#D8D8D8" stroke-width="2" fill="transparent"/>
  </symbol>
  <symbol width="18" height="18" viewBox="0 0 18 18" id="checked">
    <circle cx="9" cy="9" r="9" fill="#6139F6"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M4.75691 7.99955C4.36639 8.39008 4.36639 9.02324 4.75691 9.41377L7.58534 12.2422C7.97586 12.6327 8.60903 12.6327 8.99955 12.2422L13.2422 7.99955C13.6327 7.60903 13.6327 6.97586 13.2422 6.58534C12.8517 6.19481 12.2185 6.19481 11.828 6.58534L8.29245 10.1209L6.17112 7.99955C5.7806 7.60903 5.14744 7.60903 4.75691 7.99955Z" fill="white"/>
  </symbol>
</svg>
```

Icon.tsx
```js
import React, { SVGProps } from 'react';

import RadioButton from '@/assets/icons/radio-button.svg';

const RadioButtonIcon = ({
  checked,
  ...props
}: SVGProps<SVGSVGElement> & { checked?: boolean }) => {
  return (
    <svg width="18" height="18" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg" {...props}>
      <use href={`${RadioButton}#${checked ? 'checked' : 'unchecked'}`} />
    </svg>
  );
};

export default RadioButtonIcon;
```

Could place all checkbox svg under a same svg file, and use the id to identify the checkbox status

Reply:
But I want to keep the native "input" behavior, so is not easy to get its "value", I have to use css selectors.

Reply:
You don't have to wrap the svg with a `span`. Actually the props can be passed into svg. E.g.
```
<Check className={styles.icon} />
```

Reply:
I mean just for the SVG elements, using the id to get the desired SVG element instead of listing all of them.  Definitely we could keep the input element.

Reply:
Cool, updated, @simeng-li  PTAL